packages/lexical/src/LexicalSelection.ts

```diff
@@ -92,6 +92,45 @@ import {
 } from './LexicalUtils';
 import {$createTabNode, $isTabNode} from './nodes/LexicalTabNode';
 
+/**
+ * Options for the getComposedRanges() method. Allows specifying which
+ * shadow roots should be included when computing composed ranges.
+ */
+export type GetComposedRangesOptions = {
+  shadowRoots?: ShadowRoot[];
+};
```

Comment:
interfaces generally work better in typescript (a lot of old types in this repo use `type`, but I think that's because the authors were more familiar with Flow)

```suggestion
export interface GetComposedRangesOptions {
  shadowRoots?: ShadowRoot[];
}
```